src/routes/cylinder-set/selectCylinderSetsByOwner.ts

```diff
@@ -0,0 +1,52 @@
+import { FastifyInstance, FastifyReply, FastifyRequest } from 'fastify';
+import { knexController } from '../../database/database';
+import { Type } from '@sinclair/typebox';
+
+import {
+  CylinderSet,
+  cylinderSet,
+  CylinderSetOwnerParamsPayload,
+} from '../../types/cylinderSet.types';
+import selectCylinderSets from '../../lib/selectCylinderSets';
+import { errorHandler } from '../../lib/errorHandler';
+
+const schema = {
+  description: 'Selects diving cylinder sets with given owner',
+  tags: ['Cylinder set'],
+  response: {
+    200: Type.Array(cylinderSet),
+    401: { $ref: 'error' },
+    403: { $ref: 'error' },
+  },
+};
+
+const handler = async (
+  req: FastifyRequest<{
+    Params: CylinderSetOwnerParamsPayload;
+  }>,
+  reply: FastifyReply
+): Promise<void> => {
+  await knexController.transaction(async (trx) => {
+    const resultBody: CylinderSet[] | undefined = await selectCylinderSets(trx);
+    if (resultBody !== undefined) {
+      const result = resultBody.filter(
+        (cylinder) => cylinder.owner === req.params.cylinderSetOwner
+      );
+
+      if (result.length > 0) {
+        await reply.code(200).send(result);
+      } else {
+        await errorHandler(reply, 404, 'Cylinder set not found');
+      }
+    }
+  });
+};
+
+export default async (fastify: FastifyInstance): Promise<void> => {
+  fastify.route({
+    method: 'GET',
+    url: '/owner/:cylinderSetOwner',
```

Comment:
Entäs jos osoite olisikin / ja cylinderSetOwner annettaisiin query parametrina? Onhan tämä nimittäin vain settejen queryamista, mutta tietyn parametrin ehtojen mukaisesti. Tällöin tämä endpoint olisikin itseasiassa sama kuin selectCylinderSets johon lisätään kyseinen query parametri, eli voidaan poistaa tämä tiedosto ja implementoida toiminallisuus osaksi selectCylinderSets endpointia =)

Reply:
Sellainen endpoint on jo yksittäisen pullosetin hakemisen käytössä. (eriniminen query parametri ei vaikuta asiaan)
```
message: "Error starting server! Method 'GET' already declared for route '/api/cylinder-set/:' with constraints '{}'",
stack: "Error: Method 'GET' already declared for route '/api/cylinder-set/:' with constraints '{}'\n" +
```
Vaihtoehtoisesti voisi tehdä sillein, että jos selectCylinderSet ei löydä pullosettiä sen id:n mukaan, se sitten alkaisi etsimään omistajan mukaan.

Reply:
Niin siis poistetaan tämä `selectCylinderSetByOwner.ts` tiedosto ja tuodaan sen toiminallisuus `selectCylinderSets.ts` tiedostoon. Eli tähän selectCylinderSets handleriin implementoidaan query parametri cylinderSetOwner joka on vapaaehtoinen. Kun se on annettu, haetaan kannasta vain ne pullosetit, joiden omistaja on annettu.